File: backend/app/schemas/category.py
# ────────────────────────────────────────────────────────────────
# Путь: backend/app/schemas/category.py
# Описание: Pydantic-схемы Category.
#           Контракты запросов и ответов API.
# ────────────────────────────────────────────────────────────────

from __future__ import annotations
from datetime import datetime
from pydantic import BaseModel, ConfigDict, Field

class CategoryBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    description: str | None = Field(default=None, max_length=1000)

class CategoryCreate(CategoryBase):
    pass

class CategoryUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = Field(default=None, max_length=1000)

class CategoryRead(CategoryBase):
    model_config = ConfigDict(from_attributes=True)
    id: int
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/category.ts
//────────────────────────────────────────
// frontend/src/types/category.ts
// Типы данных Category, которые frontend получает от backend.
//────────────────────────────────────────

export type Category = {
  id: number
  name: string
  description: string | null
  created_at: string
  updated_at: string
}

export type CategoryCreatePayload = {
  name: string
  description: string | null
}